src/session/mod.rs

```diff
@@ -40,9 +39,9 @@ mod imp {
         pub(super) me: WeakRef<User>,
         pub(super) chat_list: OnceCell<ChatList>,
         pub(super) user_list: OnceCell<UserList>,
-        pub(super) basic_group_list: OnceCell<BasicGroupList>,
-        pub(super) supergroup_list: OnceCell<SupergroupList>,
-        pub(super) secret_chat_list: OnceCell<SecretChatList>,
+        pub(super) basic_group_list: RefCell<HashMap<i64, BasicGroup>>,
+        pub(super) supergroup_list: RefCell<HashMap<i64, Supergroup>>,
+        pub(super) secret_chat_list: RefCell<HashMap<i32, SecretChat>>,
```

Comment:
It may be worth naming these to something different as `_list`, since they're not lists anymore.

Reply:
Just `*_map` or plural like `basic_groups`.